tools/nni_cmd/nnictl_utils.py

```diff
@@ -736,3 +736,165 @@ def search_space_auto_gen(args):
+        exit(1)
+    if args.logDir:
+        if not os.path.exists(args.logDir):
+            print_error('Invalid: logDir path does not exist!')
+            exit(1)
+    experiment_temp_dir = os.path.join(temp_root_dir, 'experiment')
+    if not os.path.exists(os.path.join(experiment_temp_dir, 'db')):
+        print_error('Invalid archive file: db file does not exist!')
+        shutil.rmtree(temp_root_dir)
+        exit(1)
+    nnictl_temp_dir = os.path.join(temp_root_dir, 'nnictl')
+    if not os.path.exists(os.path.join(nnictl_temp_dir, '.experiment')):
+        print_error('Invalid archive file: nnictl metadata file does not exist!')
+        shutil.rmtree(temp_root_dir)
+        exit(1)
+    try:
+        with open(os.path.join(nnictl_temp_dir, '.experiment'), 'r') as file:
+            experiment_metadata = json.load(file)
+    except ValueError as err:
+        print_error('Invalid nnictl metadata file: %s' % err)
+        shutil.rmtree(temp_root_dir)
+        exit(1)
+    experiment_config = Experiments()
+    experiment_dict = experiment_config.get_all_experiments()
+    experiment_id = experiment_metadata.get('id')
+    if experiment_id in experiment_dict:
+        print_error('Invalid: experiment id already exist!')
+        shutil.rmtree(temp_root_dir)
+        exit(1)
+    if not os.path.exists(os.path.join(nnictl_temp_dir, experiment_metadata.get('fileName'))):
+        print_error('Invalid: experiment metadata does not exist!')
+        shutil.rmtree(temp_root_dir)
+        exit(1)
+
+    # Step2. Copy nnictl metadata
+    src_path = os.path.join(nnictl_temp_dir, experiment_metadata.get('fileName'))
+    dest_path = os.path.join(NNICTL_HOME_DIR, experiment_metadata.get('fileName'))
+    if os.path.exists(dest_path):
+        shutil.rmtree(dest_path)
+    shutil.copytree(src_path, dest_path)
+
+    # Step3. Copy experiment data
+    nni_config = Config(experiment_metadata.get('fileName'))
+    nnictl_exp_config = nni_config.get_config('experimentConfig')
+    if args.logDir:
+        logDir = args.logDir
+        nnictl_exp_config['logDir'] = logDir
+    else:
+        if nnictl_exp_config.get('logDir'):
+            logDir = nnictl_exp_config['logDir']
+        else:
+            logDir = NNI_HOME_DIR
+    os.rename(os.path.join(temp_root_dir, 'experiment'), os.path.join(temp_root_dir, experiment_id))
+    src_path = os.path.join(os.path.join(temp_root_dir, experiment_id))
+    dest_path = os.path.join(os.path.join(logDir, experiment_id))
+    if os.path.exists(dest_path):
+        shutil.rmtree(dest_path)
+    shutil.copytree(src_path, dest_path)
+
+    # Step4. Copy code dir
+    codeDir = os.path.expanduser(args.codeDir)
+    if not os.path.isabs(codeDir):
+        codeDir = os.path.join(os.getcwd(), codeDir)
+        print_normal('Expand codeDir to %s' % codeDir)
+    nnictl_exp_config['trial']['codeDir'] = codeDir
+    archive_code_dir = os.path.join(temp_root_dir, 'code')
```

Comment:
why not copy entire 'code'  dir using shutil.copytree?

Reply:
`shutil.copytree` only support copy entire folder to another folder, does not support merging two folder contents. We need to copy the contents under `code` folder, and do not change target folder name.